### PR TITLE
DHFPROD-4285: Can now map multiple properties of the same entity type

### PR DESCRIPTION
**Pull request by rjrudin**
"m:entity" now refers to a unique property path instead of an entity type title.

### CODE REVIEW

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mapping/entity-services/lib.sjs [hunk 1]
```diff
@@ -141,16 +167,34 @@ function buildMapProperties(mapping, entityModel) {
...
   if (dhMappingTraceIsEnabled) {
     xdmp.trace(dhMappingTrace, `Getting related mappings for '${xdmp.describe(mapping)}'`);
   }
-  let internalMappings = mapping.xpath('/properties//object-node()[exists(targetEntityType) and exists(properties)]');
-  for (let internalMapping of internalMappings) {
-    related.push(internalMapping);
-  }
-  return related;
+  Object.keys(mapping.properties).forEach(propertyTitle => {
+    const property = mapping.properties[propertyTitle];
```
Comment (bsrikan) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mapping/entity-services/lib.sjs]: We have a failing test. null check required here.

`com.marklogic.client.FailedRequestException: Local message: failed to apply resource at documents: Internal Server Error. Server Message: JS-JAVASCRIPT: Object.keys(mapping.properties).forEach(propertyTitle => { -- Error running JavaScript request: TypeError: Cannot convert undefined or null to object . See the MarkLogic server error log for further detail.`

Comment (rjrudin) [reply marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mapping/entity-services/lib.sjs]: I'll fix it, but it's really due to an invalid test file - that test is generating a mapping file with "null" as the value of "properties", which doesn't make any sense.

**Comment by rjrudin** [comment]
Fixed the failing test, which had an invalid mapping file, by ensuring that the mapping passed into getObjectPropertyMappings has a non-null "properties" property.

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mapping/entity-services/lib.sjs (hunk 1 above)
Comment (bsrikan) [reply marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mapping/entity-services/lib.sjs]: True that about the mapping file in the test. When a mapping step is created a version 0 file has similar structure ie no properties.

**Pull request merged**